tests/test_markers.py

```diff
@@ -96,51 +94,6 @@ def test_fails_when_undefined(self):
 
 
 class TestDefaultEnvironment:
-    @pytest.mark.skipif(
-        hasattr(sys, "implementation"), reason="sys.implementation does exist"
-    )
-    def test_matches_expected_no_sys_implementation(self):
-        environment = default_environment()
-
-        assert environment == {
-            "implementation_name": "",
-            "implementation_version": "0",
-            "os_name": os.name,
-            "platform_machine": platform.machine(),
-            "platform_release": platform.release(),
-            "platform_system": platform.system(),
-            "platform_version": platform.version(),
-            "python_full_version": platform.python_version(),
-            "platform_python_implementation": platform.python_implementation(),
-            "python_version": ".".join(platform.python_version_tuple()[:2]),
-            "sys_platform": sys.platform,
-        }
-
-    @pytest.mark.skipif(
-        not hasattr(sys, "implementation"), reason="sys.implementation does not exist"
-    )
-    def test_matches_expected_deleted_sys_implementation(self, monkeypatch):
-        monkeypatch.delattr(sys, "implementation")
-
-        environment = default_environment()
-
-        assert environment == {
-            "implementation_name": "",
-            "implementation_version": "0",
-            "os_name": os.name,
-            "platform_machine": platform.machine(),
-            "platform_release": platform.release(),
-            "platform_system": platform.system(),
-            "platform_version": platform.version(),
-            "python_full_version": platform.python_version(),
-            "platform_python_implementation": platform.python_implementation(),
-            "python_version": ".".join(platform.python_version_tuple()[:2]),
-            "sys_platform": sys.platform,
-        }
```

Comment:
This should not be removed?

Reply:
The behaviour this is testing has been removed, and the main behaviour is tested in the next test.

Reply:
If I understand the test correctly, it is deleting the `sys.implementation` attribute in an attempt to act like Python 2 which does not have the attribute. Now that Python 2 is not supported, this seems like a non-issue to me and not something that should be supported.

After all, we don't support deleting the attribute `collections.defaultdict` or `collections.namedtuple`. Why should this be a special case?

So, IMO, it should be deleted. If there is still a desire to keep it, let me know and I'll restore it with the code in `packaging/tags.py`.